metals/src/main/scala/scala/meta/internal/metals/codeactions/ConvertCommentCodeAction.scala

```diff
@@ -0,0 +1,147 @@
+package scala.meta.internal.metals.codeactions
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.Future
+
+import scala.meta.internal.metals.Buffers
+import scala.meta.internal.metals.MetalsEnrichments._
+import scala.meta.internal.metals.codeactions.CodeAction
+import scala.meta.internal.metals.codeactions.MillifyScalaCliDependencyCodeAction._
+import scala.meta.internal.parsing.Trees
+import scala.meta.io.AbsolutePath
+import scala.meta.pc.CancelToken
+import scala.meta.tokens.Token
+import scala.meta.tokens.Tokens
+
+import org.eclipse.{lsp4j => l}
+
+class ConvertCommentCodeAction(buffers: Buffers) extends CodeAction {
+
+  override def kind: String = l.CodeActionKind.RefactorRewrite
+
+  override def contribute(params: l.CodeActionParams, token: CancelToken)(
+      implicit ec: ExecutionContext
+  ): Future[Seq[l.CodeAction]] = Future {
+    val path = params.getTextDocument().getUri().toAbsolutePath
+    val range = params.getRange()
+    val couldBeScalaCli = path.isScalaScript || path.isScala
+
+    (for {
+      content <- buffers.get(path)
+      tokens <- tokenizeIfNotRangeSelection(range, content)
+      contentLines = content.split("\\r?(\\n|\\r)").toVector
+      codeAction <- createActionIfPossible(
+        path,
+        tokens,
+        range,
+        isSingleLineComment(contentLines, couldBeScalaCli),
+      )
+    } yield codeAction).toList
+  }
+
+  private def tokenizeIfNotRangeSelection(range: l.Range, content: String) = {
+    Option
+      .when(range.getStart == range.getEnd)(
+        Trees.defaultTokenizerDialect(content).tokenize
+      )
+      .flatMap(_.toOption)
+  }
+
+  private def isSingleLineComment(
+      contentLines: Vector[String],
+      couldBeScalaCli: Boolean,
+  )(
+      t: Token
+  ): Boolean = t match {
+    case tc: Token.Comment =>
+      val currentLine = contentLines(t.pos.startLine)
+      val isLikelyScalaCliDirective =
+        couldBeScalaCli && isScalaCliUsingDirectiveComment(currentLine)
+      val tokenIsSingleLine = tc.pos.startLine == tc.pos.endLine
+      val tokenStartsWithDoubleSlash =
+        currentLine.slice(tc.pos.startColumn, tc.pos.startColumn + 2) == "//"
+
+      !isLikelyScalaCliDirective &&
+      tokenIsSingleLine &&
+      tokenStartsWithDoubleSlash
+    case _ => false
+  }
+
+  private def createActionIfPossible(
+      path: AbsolutePath,
+      tokens: Tokens,
+      range: l.Range,
+      isSingleLineComment: Token => Boolean,
+  ): Option[l.CodeAction] = {
+    val indexOfLineTokenUnderCursor = tokens.lastIndexWhere(t =>
+      t.pos.startLine == range.getStart.getLine
+        && t.pos.endLine == t.pos.startLine
+        && t.pos.startColumn < range.getStart.getCharacter
```

Comment:
```suggestion
    val indexOfLineTokenUnderCursor = tokens.indexWhere(t =>
     t.pos.encloses(range)
```
?

Reply:
Nice, but this slightly changes the behavior. 
Previously, 
```
<<>>>// comment
```
would not emit any action, but now it does as the `encloses`operation is inclusive in the terms of startColumn. 

I think that is fine, but wanted to highlight that.